src/models/models.py

```diff
@@ -32,20 +40,28 @@ class Expense(Base):
 class User(Base):
     __tablename__ = 'users'
 
-    id = mapped_column(Integer, primary_key=True, autoincrement=True)
-    auth_provider_id = mapped_column(String, unique=True)
-    username = mapped_column(String)
+    id = mapped_column(Integer, primary_key=True, autoincrement=True, nullable=False)
+    auth_provider_id = mapped_column(String, unique=True, nullable=False)
+    username = mapped_column(String, nullable=False)
     email = mapped_column(String, unique=True)
     created_at = mapped_column(DateTime, server_default=func.now())
 
+    @validates('id', 'auth_provider_id', 'username')
+    def empty_string_to_null(self, key, value):
+        return None if value == '' else value
+
     def __repr__(self) -> str:
         return f"User(id={self.id!r}, username={self.username!r}, email={self.email!r}, created_at={self.created_at.strftime('%Y-%m-%d %H:%M:%S')})"
 
 class AccessToken(Base):
     __tablename__ = 'accesstokens'
 
-    id = mapped_column(Integer, primary_key=True, autoincrement=True)
-    registered_to_user = mapped_column(ForeignKey('users.id'), ForeignKey('users.id'))
-    created_at = mapped_column(DateTime, server_default=func.now())
-    expires = mapped_column(DateTime)
-    token_value = mapped_column(String)
+    id = mapped_column(Integer, primary_key=True, autoincrement=True, nullable=False)
+    registered_to_user = mapped_column(ForeignKey('users.id'), ForeignKey('users.id'), nullable=False)
+    created_at = mapped_column(DateTime, server_default=func.now(), nullable=False)
+    expires = mapped_column(DateTime, nullable=False)
+    token_value = mapped_column(String, nullable=False)
+
+    @validates('id', 'registered_to_user ', 'created_at', 'expires', 'token_value')
```

Comment:
I wonder if it would be possible to put this on the base class... but then I don't know how we would feed it the right properties

Reply:
yeah I also didn't love copy/pasting the same function 3 times, but couldn't really think of a great way to do it.

Reply:
How about a class function on the base class and then each @validates function just simply calls that one. At least it reduces overhead when applying it to other subclasses

Reply:
But given that it's only one line................................................................

Reply:
It's also a decorator function, which tbh, I don't really know how they work.